File: apps/server/src/modules/dashboard/dashboard.controller.ts
import type { Request, Response } from 'express';
import { dashboardService } from './dashboard.service.js';
import { asyncHandler } from '../../utils/asyncHandler.js';
import { ok } from '../../utils/ApiResponse.js';

export const dashboardController = {
  admin: asyncHandler(async (_req: Request, res: Response) => {
    const data = await dashboardService.adminSummary();
    return ok(res, data, 'Admin dashboard summary');
  }),

  merchant: asyncHandler(async (req: Request, res: Response) => {
    const data = await dashboardService.merchantSummary(req.user!.id);
    return ok(res, data, 'Merchant dashboard summary');
  }),

  warehouse: asyncHandler(async (_req: Request, res: Response) => {
    const data = await dashboardService.warehouseSummary();
    return ok(res, data, 'Warehouse dashboard summary');
  }),

  finance: asyncHandler(async (_req: Request, res: Response) => {
    const data = await dashboardService.financeSummary();
    return ok(res, data, 'Finance dashboard summary');
  }),

  customer: asyncHandler(async (req: Request, res: Response) => {
    const data = await dashboardService.customerSummary(req.user!.id);
    return ok(res, data, 'Customer dashboard summary');
  }),

  support: asyncHandler(async (_req: Request, res: Response) => {
    const data = await dashboardService.supportSummary();
    return ok(res, data, 'Support dashboard summary');
  }),

  driver: asyncHandler(async (req: Request, res: Response) => {
    const data = await dashboardService.driverSummary(req.user!.id);
    return ok(res, data, 'Driver dashboard summary');
  }),
};

File: apps/guzo-api/src/main/java/et/guzo/web/controller/DashboardController.java
package et.guzo.web.controller;

import et.guzo.common.ApiResponse;
import et.guzo.security.SecurityUtil;
import et.guzo.service.DashboardService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/dashboard")
@RequiredArgsConstructor
public class DashboardController {

    private final DashboardService dashboardService;

    @GetMapping("/admin")
    public ApiResponse<Map<String, Object>> admin() {
        return ApiResponse.ok(dashboardService.adminSummary());
    }

    @GetMapping("/merchant")
    public ApiResponse<Map<String, Object>> merchant() {
        return ApiResponse.ok(dashboardService.merchantSummary(SecurityUtil.requireUser().getId()));
    }

    @GetMapping("/customer")
    public ApiResponse<Map<String, Object>> customer() {
        return ApiResponse.ok(dashboardService.customerSummary(SecurityUtil.requireUser().getId()));
    }

    @GetMapping("/driver")
    public ApiResponse<Map<String, Object>> driver() {
        return ApiResponse.ok(dashboardService.driverSummary(SecurityUtil.requireUser().getId()));
    }

    @GetMapping("/warehouse")
    public ApiResponse<Map<String, Object>> warehouse() {
        return ApiResponse.ok(dashboardService.warehouseSummary());
    }

    @GetMapping("/finance")
    public ApiResponse<Map<String, Object>> finance() {
        return ApiResponse.ok(dashboardService.financeSummary());
    }

    @GetMapping("/support")
    public ApiResponse<Map<String, Object>> support() {
        return ApiResponse.ok(dashboardService.supportSummary());
    }

    @GetMapping("/branch")
    public ApiResponse<Map<String, Object>> branch(@RequestParam String branchId) {
        return ApiResponse.ok(dashboardService.branchSummary(branchId, SecurityUtil.requireUser()));
    }
}
